### ISSUE TITLE
[doc] enhance CONTRIBUTE.md and starter doc 

### ISSUE THREAD

**Issue by chunyu3** [Azure/azure-sdk-for-net#36348]
We need to enhance CONTRIBUTE.md and starter doc:

- Add a CONTRIBUTE.md doc in the first line of starter doc, e.g.  'If you does not familiar with the .NET SDK package, please refer to [Contribute guild] '
- Add how to contribute Test, samples and so on to CONTRIBUTE.md

**Comment by m-nash** [comment]
@chunyu3 your first bullet we should do for sure.  Can you split the second bullet into a separate issue with a proposal?  I agree that we should be as concise as possible but a proposal would help illustrate the gap for everyone so we can then talk about 1) do we agree on the gap and 2) where should we fill the gap

**Comment by pallavit** [comment]
I am also curious - is this coming from any specific feedback?

**Comment by chunyu3** [comment]
> @chunyu3 your first bullet we should do for sure. Can you split the second bullet into a separate issue with a proposal? I agree with @jsquire that we should be as concise as possible but a proposal would help illustrate the gap for everyone so we can then talk about 1) do we agree on the gap and 2) where should we fill the gap

Thanks @m-nash I will file another issue for contribute.md with proposal.

**Comment by m-nash** [comment]
> I am also curious - is this coming from any specific feedback?

Yes recently the content safety team gave us some feedback on our onboarding docs and identified a few areas that they felt could be improved.
